models/organization.go

```diff
@@ -0,0 +1,218 @@
+		"organization", user.Organization,
+	).Error; err != nil {
+		return nil, err
+	}
+	// update their tier to white-labeled
+	// which will enable organizational based billing
+	if err := NewUsageManager(om.DB).UpdateTier(
+		username,
+		WhiteLabeled,
+	); err != nil {
+		return nil, err
+	}
+	// find organization model
+	org, err := om.FindByName(name)
+	if err != nil {
+		return nil, err
+	}
+	// update organization registered users
+	org.RegisteredUsers = append(org.RegisteredUsers, username)
+	// save updated org model model
+	if err := om.DB.Model(org).Update(
+		"registered_users",
+		org.RegisteredUsers,
+	).Error; err != nil {
+		return nil, err
+	}
+	return user, nil
+}
+
+// FindByName finds an organization by name
+func (om *OrgManager) FindByName(name string) (*Organization, error) {
+	org := &Organization{}
+	if err := om.DB.Where(
+		"name = ?",
+		name,
+	).First(&org).Error; err != nil {
+		return nil, err
+	}
+	return org, nil
+}
+
+// GetOrgUsers is used toreturn the registered users an organization has
+func (om *OrgManager) GetOrgUsers(name string) ([]string, error) {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return nil, err
+	}
+	return org.RegisteredUsers, nil
+}
+
+// IncreaseAccountBalance increases the amount owed by this account
+func (om *OrgManager) IncreaseAccountBalance(name string, amount float64) error {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return err
+	}
+	if org.AccountBalance+amount < org.AccountBalance {
+		return errors.New("account balance overflow error")
+	}
+	org.AccountBalance = org.AccountBalance + amount
+	return om.DB.Model(org).Update("account_balance", org.AccountBalance).Error
+}
+
+// DecreaseAccountBalance decreases the amount owed by this account
+func (om *OrgManager) DecreaseAccountBalance(name string, amount float64) error {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return err
+	}
+	if org.AccountBalance-amount > org.AccountBalance {
+		return errors.New("account balance overflow error")
+	}
+	org.AccountBalance = org.AccountBalance - amount
+	return om.DB.Model(org).Update("account_balance", org.AccountBalance).Error
+}
+
+// GetTotalStorageUsed returns the total storage in bytes consumed
+// by the organization.
+func (om *OrgManager) GetTotalStorageUsed(name string) (uint64, error) {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return 0, err
+	}
+	var total uint64
+	for _, user := range org.RegisteredUsers {
+		usage, err := NewUsageManager(om.DB).
+			FindByUserName(user)
+		if err != nil {
+			return 0, err
+		}
+		if total+usage.CurrentDataUsedBytes < total {
+			return 0, errors.New("overflow error")
+		}
+		total = total + usage.CurrentDataUsedBytes
+	}
+	return total, nil
+}
+
+// BillingReport contains a summary
+// of an organizations entire active
+// user base in the last 30 days along with
+// the USD value currently owned by the account
+type BillingReport struct {
+	Name  string        `json:"name"`
```

Comment:
it's not clear if this is the report's name or org name

Reply:
Good idea